src/pal/src/misc/sysinfo.cpp

```diff
@@ -239,6 +241,58 @@ GetSystemInfo(
     PERF_EXIT(GetSystemInfo);
 }
 
+// Get memory size multiplier based on the passed in units (k = kilo, m = mega, g = giga)
+static uint64_t GetMemorySizeMultiplier(char units)
+{
+    switch(units)
+    {
+        case 'g':
+        case 'G': return 1024 * 1024 * 1024;
+        case 'm':
+        case 'M': return 1024 * 1024;
+        case 'k':
+        case 'K': return 1024;
+    }
+
+    // No units multiplier
+    return 1;
+}
+
+#ifndef __APPLE__
```

Comment:
Isn't it better to use platform-specific #ifdefs here? This should be built on FreeBSD as well, where it's going to fail during runtime (and when it's time to add support to other OSes, it's easier to have the preprocessor #error on an #else).

Reply:
It seems that procfs exists on other Unix systems too. And here we just try once and then fall back to using sysconf, so we don't fail.